builder/bundleset.go

```diff
@@ -114,46 +125,150 @@ func sortBundles(bundles bundleSet) ([]*bundle, error) {
 
 }
 
+// ValidationLevel represents a specific validation level
+type ValidationLevel int
+
+// Enum of available validation levels
+const (
+	StrictnessLow ValidationLevel = iota
+	StrictnessMed
+	StrictnessHigh
```

Comment:
Optional: I don't think the low/med/high are worth to have. If an extra level of check is needed, start with `--strict`, and choose some meaningful properties to be there. I think in general all those properties are optional for the pipeline, but I'm not familiar with the long term plan for those.

Reply:
I agree with @cmarcelo here, if we're vetting then go all in and make it all compliant. There's no benefit to having more simply written bundles, especially if we're ever to change the format to something else - i.e json. There will be either a working definition or an unacceptable one.

Reply:
I originally had two levels -- basic and strict -- where (on top of basic pars-ability) basic checked that the bundle header Title was existent, valid, and matched the bundle file name, and strict additionally checked that all of the other bundle header fields were non-empty.

The reason I split it out into a third level is that quite a few upstream bundles have a blank Capabilities header, and were thus coming back as invalid. Most, it seems. I'm not actually sure what this header was intended for (the `games` bundle, for example, has `# [CAPABILITIES]: Makes you forget about work, for a little bit`), so it probably should just be removed if it's not meaningful and most bundles don't use it. But it looked bad saying most of our bundles are invalid. 😜

Reply:
My view of a strict vs basic is basic should just check if they are valid at all (have all the *required* fields). Strict can then check the remaining headers. In my opinion it is ok to fail strict validation on bundles with empty optional headers. I don't see any issues with upstream bundles failing strict validation if they pass basic validation, since basic is the one that matters.

I agree with @cmarcelo and @tmarcu that three levels is too much. Users will need to know too much about bundle format and what is optional/required/semi-optional to be able to realistically pick the level they want to validate at.

Reply:
Gotcha. Like I said, that's how I originally implemented it too. Happy to go back, as long as people are happy to have upstream bundles fail strict.

Reply:
Fixed in new push.

Reply:
> as long as people are happy to have upstream bundles fail strict.

Yep, I have no issues with this. We can either ignore it or take it as an AR to update the bundle headers.